### ISSUE TITLE
How to change the ARN from role to user in a hybrid published application

### ISSUE THREAD

**Issue by hamduc7** [amzn/selling-partner-api-docs#780]
Because role "IAM ARN" does not work so I need to change to user "IAM ARN" but this field is locked
![image](https://user-images.githubusercontent.com/32189853/119505148-dfa7eb00-bd96-11eb-8395-19b5c4a884ce.png)

Can anyone help?

Creating new application is not a solution

**Comment by hamduc7** [comment]
Amazon document says that we should use IAM Role here but it will not work. I have tested by creating a new app that uses IAM User and it works perfectly.
So I need a way to change this IAM ARN from role to user

**Comment by iLoveSarabear** [comment]
Same. I'd like to know as well.

**Comment by hamduc7** [comment]
Hi All, if you are using Role ARN like me then there is nothing wrong, the reason why we got "Permission Denied" because we use it wrongly. I think the document does not write well to tell us this case.

If we use User ARN then when calling SP-API we use:
1. AccessID, SecretID: from the AWS User
2. AccessToken

If we use Role ARN then we need to perform an extending step: 
1. call AssumeRole API to get the keys using:

- AccessID, SecretID: from the AWS User
- RoleARN: this is the role ARN we created
=>This function will return: AccessID, SecretID and SessionToken
2. When calling SP-API we MUST use the keys returned from step#1 above. For example, the new AccessID & SecretID (not the AccessID from AWS User). You also need to pass both: **x-amz-access-token** = the normal access_token and **x-amz-security-token** = the SessionToken in step#2

I was stuck with this for some days so I would like to share!

By the way, using User ARN is easier. Not sure why Amazon recommends using Role ARN.

Le
